File: benchmarks/tsb/bench_timedelta_ops_na.ts
/**
 * Benchmark: toTimedelta / formatTimedelta / parseFrac — timedelta parsing and formatting.
 * Outputs JSON: {"function": "timedelta_ops_na", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { toTimedelta, formatTimedelta, parseFrac, Timedelta } from "../../src/index.ts";

const WARMUP = 5;
const ITERATIONS = 100;

const td = new Timedelta(3661000); // 1h 1m 1s in ms
const vals = ["1h", "30m", "2.5s", "100ms", "1d 2h"];

for (let i = 0; i < WARMUP; i++) {
  for (const v of vals) toTimedelta(v);
  formatTimedelta(td);
  parseFrac("1.5");
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  for (const v of vals) toTimedelta(v);
  formatTimedelta(td);
  parseFrac("1.5");
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "timedelta_ops_na",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_timedelta_ops_na.py
"""
Benchmark: pd.Timedelta parsing / formatting — timedelta ops.
Outputs JSON: {"function": "timedelta_ops_na", "mean_ms": ..., "iterations": ..., "total_ms": ...}
"""
import json
import time
import pandas as pd

WARMUP = 5
ITERATIONS = 100

td = pd.Timedelta(hours=1, minutes=1, seconds=1)
vals = ["1h", "30min", "2.5s", "100ms", "1D 2h"]

for _ in range(WARMUP):
    for v in vals:
        pd.Timedelta(v)
    str(td)

start = time.perf_counter()
for _ in range(ITERATIONS):
    for v in vals:
        pd.Timedelta(v)
    str(td)
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "timedelta_ops_na",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))
